plugin.rb

```diff
@@ -134,6 +135,43 @@ def execute(args)
         end
       end
     end
+
+    class DetectPostsTranslation < ::Jobs::Scheduled
+      sidekiq_options retry: false
+      every 5.minutes
+
+      BATCH_SIZE = 100
+      MAX_QUEUE_SIZE = 1000
+
+      def execute(args)
+        return unless SiteSetting.translator_enabled
+
+        post_ids = Discourse.redis.spop(DiscourseTranslator::LANG_DETECT_NEEDED, MAX_QUEUE_SIZE)
```

Comment:
Wonder if we shouldn't build a thin wrapper around these Redis operations, to avoid having all these calls everywhere.

```
class PostTranslationQueue
  def push(post_id); end
  def find_in_batch; end
  def has(post_id); end
  def remove(post_id); end
end
```

Something similar to this

Reply:
Up to you, I would argue that you are introducing this code now, as in you are not fixing existing code, so it seems like the best time to do it 🤷‍♂️